pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -299,18 +292,28 @@ private RebalanceResult doRebalance(TableConfig tableConfig, RebalanceConfig reb
               + "rebalance", rebalanceJobId, tableNameWithType), e);
       return new RebalanceResult(rebalanceJobId, RebalanceResult.Status.FAILED,
           "Caught exception while calculating target assignment: " + e, instancePartitionsMap,
-          tierToInstancePartitionsMap, null, preChecksResult, null);
+          tierToInstancePartitionsMap, null, null, null);
     }
 
     boolean segmentAssignmentUnchanged = currentAssignment.equals(targetAssignment);
     LOGGER.info("For rebalanceId: {}, instancePartitionsUnchanged: {}, tierInstancePartitionsUnchanged: {}, "
             + "segmentAssignmentUnchanged: {} for table: {}", rebalanceJobId, instancePartitionsUnchanged,
         tierInstancePartitionsUnchanged, segmentAssignmentUnchanged, tableNameWithType);
 
+    TableSizeReader.TableSubTypeSizeDetails tableSubTypeSizeDetails = fetchTableSizeDetails(tableNameWithType);
+
+    Map<String, String> preChecksResult = null;
+    if (preChecks && _rebalancePreChecker != null) {
+      // TODO: consider making an error or warning log when pre-checks are enabled but the pre-checker is not set
+      RebalancePreChecker.TableFacts tableFacts = new RebalancePreChecker.TableFacts(rebalanceJobId, tableNameWithType,
+          tableConfig, currentAssignment, targetAssignment, tableSubTypeSizeDetails);
+      preChecksResult = _rebalancePreChecker.check(tableFacts);
+    }
```

Comment:
nit: let's add an else for if `preChecks` are enabled but `_rebalancePreChecker` is null so that we can debug this scenario from logs